PerimeterXModule/Internals/Helpers/BodyReader.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System.IO;
+using System.Runtime.Remoting.Contexts;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Web;
+
+namespace PerimeterX
+{
+    public class BodyReader
+    {
+        public static string ReadRequestBodyAsync(HttpRequest request)
+        {
+            MemoryStream memstream = new MemoryStream();
+            request.InputStream.CopyTo(memstream);
+            memstream.Position = 0;
+            using (StreamReader reader = new StreamReader(memstream))
+            {
+                string text = reader.ReadToEnd();
+                request.InputStream.Position = 0;
+                return text;
+            }
```

Comment:
I like this separation. Just want to make sure -

(1) Does this support reading the body from the same request multiple times?
(2) Is the 2nd time any more efficient than the 1st time? If we need the req body for multiple reasons (CI & GQL), will we need to read the body twice and will it take twice as long as reading it just once?